core/lib/zksync_core/src/metadata_calculator/mod.rs

```diff
@@ -163,11 +166,28 @@ impl MetadataCalculator {
             "Merkle tree is initialized and ready to process L1 batches: {:?}",
             tree_reader.clone().info().await
         );
+        let (tree_pruner, pruner_handle) = tree.pruner();
+        let pruner_version_source = KeepPruningSyncedWithDbPruning {
+            pool: pool.clone(),
+            rt_handle: Handle::current(),
+        };
+        let pruner_thread = thread::spawn(move || tree_pruner.run(&pruner_version_source));
```

Comment:
Would `tokio::task::spawn_blocking` not work here?

Reply:
corrected